### ISSUE TITLE
Network read-out not implemented yet

### ISSUE THREAD

**Issue by 0323pin** [grtcdr/citron#7]
Hi @grtcdr

I've updated the `citron` package to 0.6.0 this morning. The commit freeze is still on ahead of the stable branching, so I should not merge this right now.

Anyway, just a reminder that the network read-out shows as "not implemented yet" on NetBSD. I'm aware of the reasons behind but, perhaps having it display the old "localIP" would be better than nothing.

Thank you for this new release.
Regards

**Comment by grtcdr** [comment]
This is odd, Local IP is supposed to appear even if the TX/RX bytes data is unimplemented.

**Comment by grtcdr** [comment]
The tests I've set up for the different formatters are not failing, which leads me to believe that there's a problem with the fetching of local IP addresses in libmacchina.

I can't reproduce this on Linux, but I'll have a close look at our most recent changes in libmacchina and hopefully resolve this issue.

**Comment by 0323pin** [comment]
`libmacchina` up to v4.0.2 works fine. LocalIP displays on `macchina`.

**Comment by grtcdr** [comment]
What about the latest version?

**Comment by 0323pin** [comment]
Building 0.6.2 right now, back to you soon.

**Comment by grtcdr** [comment]
Okay, thanks!

**Comment by 0323pin** [comment]
😢 https://postimg.cc/2qqYC5Jy

**Comment by grtcdr** [comment]
I'll get back to the drawing board, I guess.

**Comment by 0323pin** [comment]
Just so you know, this particular read-out is not that important to me, I usually disable it on `macchina` as well. But, I think it's important for you to know about it.

Thank you for being so cooperative.

OT: I hope the freeze ends soon, I'm starting to "pile-up" updates, 8 packages so far laying in My home directory.

**Comment by grtcdr** [comment]
> Just so you know, this particular read-out is not that important to me [...] But, I think it's important for you to know about it.

Yeah, thanks for notifying me about this. I think it's very weird that this is happening, since the `netbsd` module [does in fact](https://github.com/Macchina-CLI/libmacchina/blob/c8117814da9e470ffe2c40833f93849d7bd01e67/src/netbsd/mod.rs#L480-L482), implement the `logical_address` method.

> Thank you for being so cooperative.

Of course!

**Comment by grtcdr** [comment]
After attempting to reproduce this, I can tell you no weirdness is going on (unlike what I previously said). I'll roll out the fix very soon.

**Comment by 0323pin** [comment]
:tada: Awesome, thanks. So, the issue was on `citron` and not on `libmacchina` after all.
I'll build it in the morning, moving over to the latest head and replying from my Void laptop while burning NetBSD-9.99.93 :smile:

**Comment by grtcdr** [comment]
> :tada: Awesome, thanks. So, the issue was on citron and not on libmacchina after all.

Yeah, I was surprised. :man_technologist: 

>  I'll build it in the morning, moving over to the latest head and replying from my Void laptop while burning NetBSD-9.99.93 smile

That sounds like fun, any cool new features?

**Comment by 0323pin** [comment]
Yeah, not that it should make much of a difference to me on an Intel machine but, https://mail-index.netbsd.org/current-users/2021/12/18/msg041777.html

If the stable branching doesn't happen during the night, I'll push citron-0.6.4 as a bug fix release ;)

**Comment by grtcdr** [comment]
> Yeah, not that it should make much of a difference to me on an Intel machine but

Ah, and the changes are ARM specific too.

> If the stable branching doesn't happen during the night, I'll push citron-0.6.4 as a bug fix release ;)

Awesome, thank you!

**Comment by 0323pin** [comment]
@grtcdr v0.6.4 is working fine, thanks.
I'll try to merge the update but, I can't promise anything as I'm having serious problems after updating to 9.99.93.

So far, I've had 3 freezes followed by kernel panic and the system rebooting on it's own. I've also had to force shutdown by holding the power twice and had to do a semi-manual `fsck` to repair the file system.

I've a coredump which I need to look into but, I'm a noob when it comes to `gdb`. I want to report this, as I'm fully convinced this is related to the

**Comment by 0323pin** [comment]
@grtcdr I've merged `citron`-0.6.4 so it will be available on the stable branch.

I've reported my issues after upgrading to 9.99.93, https://mail-index.netbsd.org/current-users/2021/12/27/msg041840.html

The crashes happen everytime I start a web browser.

**Comment by grtcdr** [comment]
Fuck me that's one hell of nightmare...

I'm pretty baffled this passed through the testing phase, but I'm also not surprised since it affects a very specific use case (that's probably not taken into account by the project's testing framework).

I got to say I'm amazed at your level of patience though, if something like this were to happen to me... I'd do some things I'd definitely regret.

I hope this gets sorted as quickly as possible, and that the same doesn't happen to your Void machi-

_It's way too early for me to pull a joke like this..._

Ah before I forget, thank you for your work, as always :heart:

**Comment by 0323pin** [comment]
>  I got to say I'm amazed at your level of patience though, if something like this were to happen to me... I'd do some things I'd definitely regret.

🤣 
Like what, distrohop 😆 

I don't want to run anyother BSD.

**Comment by grtcdr** [comment]
_"My disappointment would be immeasurable, and my day would be ruined"_

**Comment by 0323pin** [comment]
But, right now I'm really **not happy**, I was trying to get a stack backtrace and the system rebooted. The it asked to manually run fsck and I fear fs corruption, it's endlessly re-assigning nodes.

Another reboot ... Another Rooth file system check and ... Failed 😢 

Where's my gparted USB...

**Comment by grtcdr** [comment]
_What the hell?!_

You should jump back on Void until the devs figure out what's making your billions of transistors go haywire. This release has been a real doozy by the looks of it.

**Comment by 0323pin** [comment]
Yeah, but that's what you risk when running untested HEAD. We are the testing team.

I won't re-install until this is sorted by I'm in touch with the guy that pushed the changes. I'm still trying to get him a stack backtrace but it's hard.
The fs can only be mounted read only and I only have access to a rescue shell. I've tried to push the 7 package updates I was holding into my git repo but, obviously I can't as the fs is read only.

I might get them back if I manage to revert a few commits from my Void machine.

Oh well, enough of that for now.

**Comment by grtcdr** [comment]
Ah, you never mentioned that it was a testing branch!

Still a bummer though :/

**Comment by 0323pin** [comment]
I'm gonna try upgrading to today's build and see if `sysinst` can fix the broken fs.
Last try before cleaning-up my ssd.

**Comment by 0323pin** [comment]
:laughing: https://codeberg.org/pin/leftwm-netbsd/commit/bdc83703747dc64d0d18a81d99a75e5f479c629a

**Comment by grtcdr** [comment]
Did it work out in the end?

**Comment by 0323pin** [comment]
> Did it work out in the end? I assume it did since you've recovered your packages somehow.

No, `sysinst` couldn't fix the mess. I took a fat32 formated usb and mounted it while in the rescue shell. Even though the fs was read only, I could copy over the packages and the debug data to the usb.

So now, I've provided the core-dumps and the debug data. Not exactly the backtrace but, with those two it's possible to get the stack backtrace on a working installation. I hope he can figure our what is causing it and commit a fix soon.

I'm also happy that my work was recovered. Unmerged packages should probably be in a git repo in the future, and not just on my local disk. At least when they are merge ready.

With these recovered, I could finally let `gparted` do it's job.
